functions/src/services/GoogleSheet.ts

```diff
@@ -0,0 +1,292 @@
+const gapi = require('googleapis').google;
+
+export enum GoogleScopes {
+  SpreadSheets = 'https://www.googleapis.com/auth/spreadsheets',
+}
+
+type ISheetId = string;
+
+// Create other SpreadSheet Interfaces and add to IProjectSpreadSheetNames
+// enum IOtherReportSheet {  }
+
+export type IProjectSpreadSheetNames = string;
+export type ISheetRowTypes = any;
+
+enum IMajorDimensions {
+  Rows = 'ROWS',
+  Columns = 'COLUMNS',
+}
+
+enum IValueInputOption {
+  USER_ENTERED = 'USER_ENTERED',
+  RAW = 'RAW',
+}
+
+interface IValueRange {
+  range: string;
+  majorDimension: IMajorDimensions;
+  values: IRowValues[];
+}
+
+interface IGetSheetResponse {
+  headers: any[];
+  rows: any[];
+}
+
+abstract class SpreadSheet {
```

Comment:
I doubt that we need this level of abstraction. Really, there is no chance we will use anything besides Google Sheets as we already have legacy system running on it. Please consider.

Reply:
Removed.